File: apps/client/src/services/agent.service.ts
/**
 * Agent服务 - 提供与Agent相关的操作方法
 * 使用 Tauri 的 invoke 调用后端 API
 */

import { invoke } from '@tauri-apps/api/core';
import { Agent, CreateAgentRequest, UpdateAgentRequest } from '../types';

class AgentService {
  /**
   * 获取所有Agent
   * @returns Agent列表
   */
  async getAllAgents(): Promise<Agent[]> {
    return await invoke<Agent[]>('get_all_agents');
  }

  /**
   * 根据ID获取Agent
   * @param id Agent ID
   * @returns Agent信息
   */
  async getAgentById(id: string): Promise<Agent> {
    return await invoke<Agent>('get_agent_by_id', { id });
  }

  /**
   * 根据用户ID获取Agent
   * @param userId 用户ID
   * @returns Agent列表
   */
  async getAgentsByUserId(userId: string): Promise<Agent[]> {
    return await invoke<Agent[]>('get_agents_by_user_id', { userId });
  }

  /**
   * 创建Agent
   * @param request 创建Agent请求
   * @returns 创建的Agent信息
   */
  async createAgent(request: CreateAgentRequest): Promise<Agent> {
    return await invoke<Agent>('create_agent', { request });
  }

  /**
   * 更新Agent
   * @param request 更新Agent请求
   * @returns 更新后的Agent信息
   */
  async updateAgent(request: UpdateAgentRequest): Promise<Agent> {
    return await invoke<Agent>('update_agent', { request });
  }

  /**
   * 删除Agent
   * @param id Agent ID
   * @returns 操作结果
   */
  async deleteAgent(id: string): Promise<boolean> {
    return await invoke<boolean>('delete_agent', { id });
  }
}

// 导出单例实例
export const agentService = new AgentService();

File: apps/client/src-tauri/src/services/agent_service.rs
use crate::db::DbPool;
use crate::models::{Agent, NewAgent};
use crate::repositories::agent_repository::AgentRepository;
use crate::repositories::error::{RepositoryError, RepositoryResult};

pub struct AgentService {
    repository: AgentRepository,
}

impl AgentService {
    pub fn new(pool: DbPool) -> Self {
        Self {
            repository: AgentRepository::new(pool),
        }
    }

    // 创建新Agent
    pub fn create_agent(
        &self,
        name: String,
        model_name: String,
        system_prompt: String,
        temperature: f32,
        max_tokens: Option<i32>,
        top_p: Option<f32>,
        avatar_url: Option<String>,
        description: Option<String>,
        is_streaming: bool,
        user_id: Option<String>,
    ) -> RepositoryResult<Agent> {
        self.repository
            .create_with_defaults(name, model_name, system_prompt, temperature, user_id)
    }

    // 获取所有Agent
    pub fn get_all_agents(&self) -> RepositoryResult<Vec<Agent>> {
        self.repository.find_all()
    }

    // 根据ID获取Agent
    pub fn get_agent_by_id(&self, id: &str) -> RepositoryResult<Agent> {
        self.repository.find_by_id(id)
    }

    // 根据用户ID获取Agent
    pub fn get_agents_by_user_id(&self, user_id: &str) -> RepositoryResult<Vec<Agent>> {
        self.repository.find_by_user_id(user_id)
    }

    // 更新Agent
    pub fn update_agent(&self, id: &str, agent: Agent) -> RepositoryResult<Agent> {
        // 检查Agent是否存在
        self.repository.find_by_id(id)?;

        self.repository.update(id, agent)
    }

    // 删除Agent
    pub fn delete_agent(&self, id: &str) -> RepositoryResult<usize> {
        // 检查Agent是否存在
        self.repository.find_by_id(id)?;

        self.repository.delete(id)
    }
}
